erpnext/patches/v14_0/change_is_subcontracted_fieldtype.py

```diff
@@ -0,0 +1,12 @@
+# Copyright (c) 2022, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+import frappe
+
+
+def execute():
+	for doctype in ["Purchase Order", "Purchase Receipt", "Purchase Invoice", "Supplier Quotation"]:
+		docs = frappe.get_all(doctype, fields=["name", "is_subcontracted"])
+		for doc in docs:
+			value = "1" if doc.is_subcontracted == "Yes" else "0"
+			frappe.db.set_value(doctype, doc.name, "is_subcontracted", value, update_modified=False)
```

Comment:
This will take a very long time. Write raw SQL / QB. 

Also patches should ideally be idempotent i.e. rerunning them shouldn't cause side effects. 



Example: 

https://github.com/frappe/erpnext/blob/f754e76067c64336a008e800e788e722efb63452/erpnext/patches/v12_0/update_is_cancelled_field.py#L19-L34